ui/src/app/pages/homepage/recent-resources.tsx

```diff
@@ -93,13 +95,19 @@ export const RecentResources = fp.flow(withCdrVersions())((props: Props) => {
   const [resources, setResources] = useState<WorkspaceResourceResponse>();
   const [wsMap, setWorkspaceMap] = useState<Map<string, Workspace>>();
   const [tableData, setTableData] = useState<TableData[]>();
+  const [apiLoadError, setApiLoadError] = useState<string>(null);
 
-  const loadResources = () => {
+  const loadResources = async () => {
     setLoading(true);
-    return userMetricsApi()
+    await userMetricsApi()
       .getUserRecentResources()
       .then(setResources)
-      .then(() => setLoading(false));
+      .catch(() => {
+        setApiLoadError(
+          'An error occurred while loading recent resources. Please refresh the page to reload.'
```

Comment:
I wonder if the error message should be like An error occurred while loading recent resources. Please refresh the page to reload or try again later'

Reply:
@jmthibault79 Any thoughts on PR or the error message?

Reply:
either is fine with me.  the logic looks good.

Reply:
Okay. I'll leave it as it is.